src/BlockTypes/ProductPrice.php

```diff
@@ -45,12 +45,44 @@ protected function get_block_type_supports() {
 	}
 
 	/**
-	 * Register script and style assets for the block type before it is registered.
-	 *
-	 * This registers the scripts; it does not enqueue them.
+	 * It is necessary to register and enqueues assets during the render phase because we want to load assets only if the block has the content.
 	 */
 	protected function register_block_type_assets() {
-		parent::register_block_type_assets();
-		$this->register_chunk_translations( [ $this->block_name ] );
+		return null;
```

Comment:
Could you tell me more about this too? Why do we need the function to be there and do nothing? Do we need to override the parent class method but have a noop?

Reply:
The latter is the case here. In the parent `AbstractBlock` class that we're inheriting from, the `register_block_type_assets` [is conditionally registering](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/BlockTypes/AbstractBlock.php#L120) a handful of scripts that we do not want to be registered in this scenario. So we're overriding that parent method and just returning `null`.

We're also taking this approach in our other blocks ([example](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/BlockTypes/ProductImage.php#L52)).

Reply:
Yeah, that sounds good to me, I'll update!  🙂

Reply:
Updated in 2b212e1.